cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerDeployClientConfigProvider.java

```diff
@@ -258,4 +265,21 @@ private void enhanceRequestParams(Map<String, List<String>> headers, List<Pair>
     ExecutorService createExecutor() {
         return Executors.newSingleThreadExecutor();
     }
+
+    @VisibleForTesting
+    void addPathSegmentsFromDefaultUrl(
+            HttpUrl defaultHttpUrl, HttpUrl.Builder httpUrlBuilder) {
+        List<String> defaultPathSegments = defaultHttpUrl.pathSegments();
+        int apiIndex = defaultPathSegments.indexOf("api");
+        int versionNumberIndex =  apiIndex == INVALID_INDEX ? INVALID_INDEX : apiIndex + 1;
+        for (int i = 0; i < defaultPathSegments.size(); i++) {
+            String pathSegment = defaultPathSegments.get(i);
+            if (i != apiIndex && i != versionNumberIndex) {
```

Comment:
although this code works well, and I think we might go with it, this solution is not self explaining when we look at the code.
like you have to understand that if apiIndex took the value of `-1` then `versionNumberIndex` would take the same value and this if condition would be always true. I think the code should be more self explaining